build/config/compiler/BUILD.gn

```diff
@@ -627,12 +630,15 @@ if (is_win) {
   default_warning_flags += [
     # Enables.
     "-Wendif-labels",  # Weird old-style text after an #endif.
-    "-Werror",  # Warnings as errors.
 
     # Disables.
     "-Wno-missing-field-initializers",  # "struct foo f = {0};"
     "-Wno-unused-parameter",  # Unused function parameters.
   ]
+  if (!is_linux || is_glibc) {
+    # Warnings as errors. Fails on musl.
+    default_warning_flags += [ "-Werror" ]
```

Comment:
Is this something we can "just fix"? Typically if absolutely necessary we disable specific warnings rather than disable warnings as errors.

Reply:
replaced with `-Wno-sign-compare` for musl only. I believe it was the only warning triggered, but I currently struggle with dart 3.x requirement to check locally